src/lib.rs

```diff
@@ -53,8 +54,17 @@ mod tests {
 
     #[async_trait::async_trait]
     impl Transport for TestTransport {
-        async fn get_json<T: DeserializeOwned>(&self, url: &str) -> Result<T> {
-            let json: String = isahc::get_async(Uri::from_str(url)?).await?.text().await?;
+        async fn get_json<T: DeserializeOwned>(
+            &self,
+            url: &str,
+            api_key: Option<String>,
```

Comment:
nit: Maybe this can be (using https://docs.rs/http/0.2.4/http/header/struct.HeaderMap.html):
```suggestion
            headers: http::header::HeaderMap,
```

Because, while BlockNative uses the `Authorization` header, others might use `X-ApiKey` or some other non-standard header for specifying API keys.

The choice of [`http`](https://docs.rs/http) is because:
- It provides general HTTP types only
- Used by `isahc`, the HTTP client we use, but also by others like `reqwest` and `hyper`.

Reply:
Nice suggestion. Fixed.